### ISSUE TITLE
vagrant provison curl ssl error

### ISSUE THREAD

**Issue by vikas027** [mitchellh/vagrant#6693]
I have vagrant-1.7.4-1 installed on a CentOS 7.1 x86_64 box. I am encountering an error while using curl with https URLs.

vagrant provision --debug shows that I am stuck here

```bash
DEBUG ssh: PTY stdout parsed: Chef: 12.5.1
 INFO downloader: Downloader starting download:
 INFO downloader:   -- Source: https://www.getchef.com/chef/install.sh
 INFO downloader:   -- Destination: /root/.vagrant.d/tmp/1450662154-install.sh
 INFO subprocess: Starting process: ["/opt/vagrant/embedded/bin/curl", "-q", "--fail", "--location", "--max-redirs", "10", "--user-agent", "Vagrant/1.7.4", "--output", "/root/.vagrant.d/tmp/1450662154-install.sh", "https://www.getchef.com/chef/install.sh"]
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stderr:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
...
...
```

This is basically the below command which also throws the same error.
```bash
 # /opt/vagrant/embedded/bin/curl -q --fail --location --max-redirs 10 --user-agent Vagrant/1.7.4 --output /root/.vagrant.d/tmp/1450656775-install.sh https://www.getchef.com/chef/install.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:01:13 --:--:--     0curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
#
```

However, it runs fine if I use `--insecure` switch
```bash
# /opt/vagrant/embedded/bin/curl --insecure --output /root/.vagrant.d/tmp/1450656775-install.sh https://www.getchef.com/chef/install.sh
OR
# /opt/vagrant/embedded/bin/curl --insecure --user-agent Vagrant/1.7.4 --output /root/.vagrant.d/tmp/1450656775-install.sh https://www.getchef.com/chef/install.sh
```
How I can make this with `vagrant provision` command ?

**Comment by vikas027** [comment]
I have got a workaround by adding `@insecure = true` in `/opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/downloader.rb` in `options` method

**Comment by sethvargo** [comment]
This will be fixed in Vagrant 1.8. I cannot find the other issues, but if you search closed issues for the 1.8 milestone, it's been fixed in the installer because we updated to the latest CA bundle.

**Comment by vikas027** [comment]
Thanks @sethvargo
